src/components/postBody/PostBody.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { Fragment, useState, useRef, useEffect } from "react";
+import { withPosts } from "../../HOC/withPosts";
+import { Link } from "react-router-dom";
+import { ThumbUp, Comment } from "@material-ui/icons";
+import { useLike } from "../../hooks/useLike";
+import { Button } from "@material-ui/core";
+import { useDeleteComment } from "../../hooks/useDeleteComment";
+import { useDeletePost } from "../../hooks/useDeletePost";
+import { withAuth } from "../../HOC/withAuth";
+const PostBodyComponent = ({ user, post, postsState, setPostsState }) => {
+  const [resLikes, setLike] = useLike();
+  const [deleteCommentPostId, setDeleteCommentPostId] = useState();
+  const [deleteComment, setDeleteComment] = useDeleteComment();
+  const [deletePost, setDeletePost] = useDeletePost();
+  const [deletePostIdState, setDeletePostIdState] = useState();
+
+  const handleLike = (id) => {
+    setLike(id);
+    localId.current = id;
```

Comment:
where is`localId ` decleared?, make sure to declare before using it also why do we need ref here ?